### PR TITLE
many: move Uname to osutil, for more DRY and easier porting.

### PR DESCRIPTION
**Pull request by chipaca**
Before this there was an Uname implementation in arch and one in
release (the implementation is slightly tricky because
syscall.Utsname's internals are arch-dependent). This drops both in
favour of a new one in osutil, in order to keep syscalls limited to
this package as much as possible.

### CODE REVIEW

File: release/uname_linux.go [hunk 1]
```diff
@@ -1,98 +0,0 @@
...
- *
- */
-
-package release
-
-import (
-	"syscall"
-)
-
-// We have to implement separate functions for the kernel version and the
```
Comment (mvo5) [inline release/uname_linux.go]: It looks like the change removes this comment - should we add it back?

Comment (chipaca) [reply release/uname_linux.go]: it adds back
```diff
+               // c can be uint8 or int8 depending on arch
```
though

File: osutil/uname_test.go [hunk 2]
```diff
@@ -17,9 +17,28 @@
...
+var _ = check.Suite(unameSuite{})
+
+func ucmd1(c *check.C, arg string) string {
+	out, err := exec.Command("uname", arg).CombinedOutput()
+	c.Assert(err, check.IsNil)
+	return string(bytes.TrimSpace(out))
+}
+
+func (unameSuite) TestUname(c *check.C) {
+	c.Check(osutil.KernelVersion(), check.Equals, ucmd1(c, "-r"))
```
Comment (zyga) [inline osutil/uname_test.go]: Neat :)

File: release/uname_linux_test.go [hunk 3]
```diff
@@ -1,73 +0,0 @@
...
-	"github.com/snapcore/snapd/release"
-)
-
-func (s *ReleaseTestSuite) TestKernelVersion(c *C) {
-	ver := release.KernelVersion()
-	// Ensure that we got something.
-	c.Check(ver, Not(Equals), "")
-}
-
-func (s *ReleaseTestSuite) TestGetKernelRelease(c *C) {
```
Comment (zyga) [inline release/uname_linux_test.go]: This seems like an useful test to keep. Should we save it?

File: release/uname_linux.go (hunk 1 above)
Comment (mvo5) [reply release/uname_linux.go]: Yeah parts are still there, the link to the upstream report and the explanation about separate functions sounded worthwhile but I have not looked too deeply so maybe its fine to drop those.

Comment (chipaca) [reply release/uname_linux.go]: yeah, I'll add it back.

File: release/uname_linux_test.go (hunk 3 above)
Comment (chipaca) [reply release/uname_linux_test.go]: done

**Pull request merged**